webui/react/src/shared/utils/routes.ts

```diff
@@ -73,5 +74,6 @@ export const routeToExternalUrl = (path: string): void => {
 };
 export const routeToReactUrl = (path: string): void => {
   logger.trace('routing to react url', path);
+  setCookie('loginRedirect', filterOutLoginLocation(window.location)?.pathname || '');
```

Comment:
**suggestion:** I feel a little nervous about using cookies in this manner because conceptually cookies are a method for the server-side to set and pass info to the client side. So there may a lot of unknown side effects that can come out of this.
My recommendation here is to create a separate `WriteableObservable` in the `auth` store just for `loginRedirect` along with a public function that can properly set the value. Much like how `auth` and `authChecked` observables are setup.